backends-velox/src/main/scala/org/apache/gluten/backendsapi/velox/VeloxListenerApi.scala

```diff
@@ -27,138 +26,67 @@ import org.apache.gluten.vectorized.{JniLibLoader, JniWorkspace}
 
 import org.apache.spark.{SparkConf, SparkContext}
 import org.apache.spark.api.plugin.PluginContext
+import org.apache.spark.internal.Logging
 import org.apache.spark.sql.execution.datasources.velox.{VeloxOrcWriterInjects, VeloxParquetWriterInjects, VeloxRowSplitter}
 import org.apache.spark.sql.expression.UDFResolver
-import org.apache.spark.sql.internal.{GlutenConfigUtil, StaticSQLConf}
-import org.apache.spark.util.SparkDirectoryUtil
+import org.apache.spark.sql.internal.GlutenConfigUtil
+import org.apache.spark.util.{SparkDirectoryUtil, SparkResourceUtil}
 
 import org.apache.commons.lang3.StringUtils
 
-import scala.sys.process._
+import java.util.concurrent.atomic.AtomicBoolean
 
-class VeloxListenerApi extends ListenerApi {
-  private val ARROW_VERSION = "1500"
+class VeloxListenerApi extends ListenerApi with Logging {
+  import VeloxListenerApi._
 
   override def onDriverStart(sc: SparkContext, pc: PluginContext): Unit = {
-    val conf = pc.conf()
-    // sql table cache serializer
-    if (conf.getBoolean(GlutenConfig.COLUMNAR_TABLE_CACHE_ENABLED.key, defaultValue = false)) {
-      conf.set(
-        StaticSQLConf.SPARK_CACHE_SERIALIZER.key,
-        "org.apache.spark.sql.execution.ColumnarCachedBatchSerializer")
+    if (!driverInitialized.compareAndSet(false, true)) {
+      // Make sure we call the static initializers only once.
+      logInfo(
+        "Skip rerunning static initializers since they are only supposed to run once." +
+          " You see this message probably because you are creating a new SparkSession.")
```

Comment:
I don’t quite understand that `onDriverStart` will be called multiple times. Can you explain?

Reply:
It happens in Spark local mode. Spark creates one driver and one executor in that mode, in the current process.

Reply:
I understand this. It should be once for `onDriverStart` and once for `onExecutorStart`, not `onDriverStart` is called twice.

Reply:
Oops, got it wrong.

`onDriverStart` will be called twice when spark session is recreated.
`onExecutorStart` may be called twice when dynamic allocation is enabled, I am not sure about this one.

Reply:
Can you give an example of how spark session is recreated, cloneSession or other?

Reply:
About `onExecutorStart`, I think it will not called twice because dynamic allocation add new executor is a new jvm.

Reply:
> Can you give an example of how spark session is recreated, cloneSession or other?

Please refer to `SparkSession.stop` or `SparkContext.stop`

Reply:
If just re-create sparkSession, it will not restart the driver. Re-creating sparkContext will restart the driver, but a new sparkConf may be set, so is it better to re-initialize it once?

Reply:
Yes, finally we should remove the flags and do re-initializations. See [my comment](https://github.com/apache/incubator-gluten/blob/4096a53e146981ff092c43b56f456137258632b9/backends-velox/src/main/scala/org/apache/gluten/backendsapi/velox/VeloxListenerApi.scala#L130-L133) and the issue https://github.com/apache/incubator-gluten/issues/6862